mkosi/initrd/__main__.py

```diff
@@ -0,0 +1,134 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import argparse
+import os
+import platform
+import shutil
+import tempfile
+from pathlib import Path
+
+from mkosi.config import OutputFormat, __version__
+from mkosi.log import log_setup
+from mkosi.run import find_binary, run, uncaught_exception_handler
+from mkosi.types import PathString
+
+
+@uncaught_exception_handler()
+def main() -> None:
+    log_setup()
+
+    parser = argparse.ArgumentParser(
+        prog="mkosi-initrd",
+        description="Build initrds or unified kernel images for the current system using mkosi",
+        allow_abbrev=False,
+        usage="mkosi-initrd [options...]",
+    )
+
+    parser.add_argument(
+        "--kernel-version",
+        metavar="KERNEL_VERSION",
+        help="Kernel version string",
+        default=platform.uname().release,
+    )
+    parser.add_argument(
+        "-t", "--format",
+        choices=[str(OutputFormat.cpio), str(OutputFormat.uki), str(OutputFormat.directory)],
+        help="Output format (CPIO archive, UKI or local directory)",
+        default="cpio",
+    )
+    parser.add_argument(
+        "-o", "--output",
+        metavar="NAME",
+        help="Output name",
+        default="initrd",
+    )
+    parser.add_argument(
+        "-O", "--output-dir",
+        metavar="DIR",
+        help="Output directory",
+        default="",
+    )
+    parser.add_argument(
+        "--debug",
+        help="Turn on debugging output",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "--version",
+        action="version",
+        version=f"mkosi {__version__}",
+    )
+
+    args = parser.parse_args()
+    output_dir = "mkosi.output" if args.output_dir == "" and Path("mkosi.output").exists() else args.output_dir
```

Comment:
Ah right because of `--directory ""` we ignore any local files. This makes me think that maybe it's simpler to not support automatic usage of `mkosi.output` and just always use the current working directory unless one is specified explicitly.

Reply:
Sure, changed.